### PR TITLE
fix: Logger settings for feature servers and updated logger for permission flow

### PR DESCRIPTION
**Pull request by redhatHameed**
# What this PR does / why we need it:

Added Logger settings for feature servers and updated logger for permission flow

# Which issue(s) this PR fixes:

# Misc

### CODE REVIEW

**Comment by dmartinol** [comment]
Shoudln't we also remove all the various call to `logger.setLevel` that we recently added to the RBAC code?
Thanks!

**Comment by redhatHameed** [comment]
> Shoudln't we also remove all the various call to `logger.setLevel` that we recently added to the RBAC code? Thanks!

done

**Review by tokoko — APPROVED** [review]
lgtm, I'll wait a bit for other comments and merge

File: sdk/python/feast/permissions/server/arrow.py [hunk 1]
```diff
@@ -49,7 +48,9 @@ def __init__(self, access_token: str, *args, **kwargs):
 
     def call_completed(self, exception):
         if exception:
-            print(f"{AuthorizationMiddleware.__name__} received {exception}")
+            logger.error(
+                f"{AuthorizationMiddleware.__name__} encountered an exception: {exception}"
```
Comment (lokeshrangineni) [inline sdk/python/feast/permissions/server/arrow.py:52]: The better way to log exception is `logging.exception` I m not sure what it logs if we do the `{exception}`.

File: sdk/python/feast/permissions/server/arrow.py [hunk 2]
```diff
@@ -69,14 +70,14 @@ def inject_user_details(context: ServerCallContext):
...
-        logger.info("No `auth` middleware.")
+        logger.warning("No `auth` middleware.")
         return
 
     sm = get_security_manager()
     if sm is not None:
         auth_middleware = cast(AuthorizationMiddleware, context.get_middleware("auth"))
         current_user = asyncio.run(auth_middleware.extract_user())
-        print(f"extracted user: {current_user}")
+        logger.info(f"User extracted: {current_user}")
```
Comment (lokeshrangineni) [inline sdk/python/feast/permissions/server/arrow.py:80]: Do we really need to log every user using info?

File: sdk/python/feast/permissions/server/grpc.py [hunk 3]
```diff
@@ -22,11 +21,13 @@ def intercept_service(self, continuation, handler_call_details):
                 metadata=dict(handler_call_details.invocation_metadata)
             )
 
-            print(f"Fetching user for token: {len(access_token)}")
+            logger.info(
+                f"Fetching user details for token of length: {len(access_token)}"
```
Comment (lokeshrangineni) [inline sdk/python/feast/permissions/server/grpc.py:25]: my 2 cents - may be debug log here is more appropriate.

**Review by dmartinol — APPROVED** [review]

File: sdk/python/feast/permissions/server/arrow.py (hunk 2 above)
Comment (redhatHameed) [reply sdk/python/feast/permissions/server/arrow.py]: I think we don't set as a debug incase if required

File: sdk/python/feast/permissions/server/grpc.py (hunk 3 above)
Comment (redhatHameed) [reply sdk/python/feast/permissions/server/grpc.py:25]: set as debug

File: sdk/python/feast/permissions/server/arrow.py (hunk 1 above)
Comment (redhatHameed) [reply sdk/python/feast/permissions/server/arrow.py:52]: set as logging.exception. These will arrow flight sever related internal exceptions.

**Pull request merged**